### PR TITLE
added events

### PR DESCRIPTION
**Pull request by Elaz**

### CODE REVIEW

File: KinEcosystem/BI/KBIEvent.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//  KinEcosystem
+//
+//  Created by Elazar Yifrach on 28/06/2018.
+//  Copyright © 2018 Kik Interactive. All rights reserved.
+//
+
+import Foundation
+
+protocol KBIEvent: Codable {
+    var common: Common { get }
```
Comment (YohayBar) [inline KinEcosystem/BI/KBIEvent.swift]: why only common? do we need user and client as well?

File: KinEcosystem/Core/Kin.swift [hunk 2]
```diff
@@ -256,4 +273,67 @@ public class Kin {
...
+        EventsStore.shared.commonProxy = CommonProxy(eventID: { () -> (String) in
+            UUID().uuidString
+        }, platform: { () -> (String) in
+            "iOS"
+        }, timestamp: { () -> (String) in
+            "\(Date().timeIntervalSince1970)"
+        }, userID: { [weak self] () -> (String) in
+            self?.core?.network.client.authToken?.ecosystem_user_id ?? ""
+            }, version: { () -> (String) in
+                "0.4.6"
```
Comment (YohayBar) [inline KinEcosystem/Core/Kin.swift]: "0.4.6" should be taken from one place

File: KinEcosystem/BI/KBIEvent.swift (hunk 1 above)
Comment (Elaz) [reply KinEcosystem/BI/KBIEvent.swift]: This is only serving a generic function. I exposed only what was needed (eventId from common)

File: KinEcosystem/BI/client/back_button_on_marketplace_page_tapped.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// Please help improve quicktype by enabling anonymous telemetry with:
+//
+//   $ quicktype --telemetry enable
```
[the remark this answers is not in the document]
Comment (Elaz) [reply KinEcosystem/BI/client/back_button_on_marketplace_page_tapped.swift]: we probably should.

File: KinEcosystem/Core/Kin.swift (hunk 2 above)
Comment (Elaz) [reply KinEcosystem/Core/Kin.swift]: true - the sdk needs a normal versioning system - will open an issue for it.

**Pull request merged**